mypy/test/helpers.py

```diff
@@ -284,3 +285,42 @@ def normalize_error_messages(messages: List[str]) -> List[str]:
     for m in messages:
         a.append(m.replace(os.sep, '/'))
     return a
+
+
+@pytest.fixture(scope='function')
+def test(request):
+    test = request.function.test
+    test.set_up()
+    request.addfinalizer(test.tear_down)
+    return test
+
+
+class PytestSuite:
+    """Assists in setting up data-driven test cases for pytest."""
+    @classmethod
+    def collect_tests(cls):
+        """
+        Sets up the child class's test case. The child must have a method
+        `cases` that returns a list of `DataDrivenTestCase`s. This method will
+        load the data-driven test cases and use setattr to assign it to the
+        class, which will allow pytest to recognize the test.
+
+        This will be called during test collection (see conftest.py in the root
+        of the repository).
+        """
+        c = cls.cases()  # type: List[DataDrivenTestCase]
```

Comment:
Let's add an implementation of `cases` on this base class that just returns an empty list. That would let us do two or three good things:
* As I remarked over on `conftest.py`, it would mean that `collect_tests` could be invoked uniformly on the base class just like on its subclasses, cutting a special case.
* It would make this function well typed! As it is, we're trying to invoke a method `cases` on a value (`cls`) whose type is `Type[PytestSuite]` -- but `PytestSuite` has no method by that name, so that method call won't type-check. You state in the docstring a requirement that derived classes possess such a method, but this is a kind of requirement that we can express quite well in the type system rather than in English; and when we can do that, we get all the benefits of the type checker in making it a requirement that we can really depend on in thinking about related code. (Which is what makes this type checker so interesting in the first place!) So let's do that.
* Let's also give both `collect_tests` and that `cases` method type annotations, so that the type-checker does its full work on them.